src/duct/core/repl.clj

```diff
@@ -17,3 +19,31 @@
   fipp/IOverride
   fipp/IEdn
   (-edn [r] (tagged-literal 'duct/resource (:path r))))
+
+(def ^:private event->file-name
+  (comp #(.getName %)
+        :file))
+
+(def ^:private matches-clojure-file?
```

Comment:
I'd prefer:

```clojure
(defn- matches-clojure-file? [filename]
  (re-matches #"^[a-zA-Z]{1}[a-zA-Z_]*(.clj|.edn)$" filename))
```

You're also missing the escapes on the periods before the extension. It should be:

```clojure
(defn- matches-clojure-file? [filename]
  (re-matches #"^[a-zA-Z]{1}[a-zA-Z_]*(\.clj|\.edn)$" filename))
```

I think I'd also be tempted to discard the beginning check for the name, and instead do:

```clojure
(defn- matches-clojure-file? [filename]
  (re-matches #".(\.clj|\.edn)$" filename))
```

So we're skipping `.clj` and `.edn` because we're testing for a single character before that, but we don't check the filename to ensure it's correct. The reason for this is that Clojure should throw an error if the file isn't found, while if we restrict the search too much, it would fail without an error.

Reply:
The reason I was checking for a single character at the start of the regular expression is because some editors create backup files during editing (for example, emacs creates .#{filename})

This would mean that `(reset)` would be called every time you edit a file, instead of when you save it.

This is of course (in my case) Emacs specific. I'm not sure how to solve this in a generic way?

Reply:
That's a sound point. What about ignoring any hidden file then? So:

```clojure
(defn- matches-clojure-file? [filename]
  (re-matches #"[^.].*\.(clj|edn)" filename))
```

I've also just realized that because we're using `re-matches` rather than `re-find`, we don't need `^` or `$`.

Reply:
Beautiful!